src/main/webapp/app/tasks/task-list/task-list.component.ts

```diff
@@ -1,28 +1,84 @@
-import { ChangeDetectionStrategy, Component, EventEmitter, Inject, Input, Output } from '@angular/core';
+import { CdkDragDrop, moveItemInArray } from "@angular/cdk/drag-drop";
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Inject,
+  Input,
+  Output,
+  ViewEncapsulation,
+} from "@angular/core";
+import { MatSnackBar } from "@angular/material/snack-bar";
+import { forkJoin } from "rxjs";
 
-import { Task } from '../task';
-import { TaskService } from '../task.service';
+import { Task } from "../task";
+import { TaskService } from "../task.service";
+import { sortTasksByStatus } from "../utils";
 
 /**
  * A list of tiny tasks.
  */
 @Component({
-  selector: 'tiny-task-list',
-  templateUrl: './task-list.component.html',
-  styleUrls: ['./task-list.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  selector: "tiny-task-list",
+  templateUrl: "./task-list.component.html",
+  styleUrls: ["./task-list.component.scss"],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TaskListComponent {
-
   @Input() tasks: Task[];
 
   @Output() deleted: EventEmitter<Task> = new EventEmitter();
+  @Output() updated: EventEmitter<Task> = new EventEmitter();
 
-  constructor(@Inject('TaskService') private taskService: TaskService) { }
+  constructor(
+    @Inject("TaskService") private taskService: TaskService,
+    private snackBar: MatSnackBar
+  ) {}
 
   delete(task: Task): void {
-    this.taskService.delete(task.id).subscribe(() => {
-      this.deleted.emit(task);
-    });
+    this.taskService.delete(task.id).subscribe(
+      () => {
+        this.deleted.emit(task);
+        this.snackBar.open("Task deleted!");
+      },
+      (error) => {
+        this.snackBar.open("Task deletion failed!");
+      }
+    );
+  }
+
+  toggleStatus(task: Task) {
```

Comment:
missing return type definition

Reply:
Yes, right. For consistency, functions should have a return type. Even though these functions are used only in the template and nowhere still, this is not an excuse.

Reply:
Definitely. I missed that.